torchok/data/datasets/base.py

```diff
@@ -60,22 +65,43 @@ def _apply_transform(self, transform: Union[BasicTransform, BaseCompose], sample
         return new_sample
 
     def _read_image(self, image_path: str) -> np.ndarray:
+        if self.reader_library == "opencv":
+            image = cv2.imread(str(image_path), cv2.IMREAD_UNCHANGED)
```

Comment:
You have to keep in mind that opencv read color images as bgr and therefore add postprocessing after reading with cv2